### ISSUE TITLE
Not such important question

### ISSUE THREAD

**Issue by OhmSnail** [keras-team/keras#11295]
`

    z_mean, z_log_var = args

    batch = K.shape(z_mean)[0]

    dim = K.int_shape(z_mean)[1]
`
K refers to tf.keras.backend

It's excerpted from keras/examples/variational_autoencoder.py, see function sampling (if you need)

> https://github.com/keras-team/keras/blob/master/examples/variational_autoencoder.py

I can't figure out why he use shape for batch, and int_shape for dim, I know shape will return a Tensor and int_shape returns a tuple.

Why don't he just use `batch, dim = K.shape(z_mean)` or `batch, dim = K.int_shape(z_mean)`, instead split it into two lines.

**Comment by yanboliang** [comment]
@OhmSnail This is because ```K.int_shape``` will return the static/referred shape, may be partially complete, the first dimension is usually ```None```. So if you want get the real value of the first dimension of a tensor shape, you should use dynamic shape ```K.shape```. The dynamic shape will be evaluated during computing. Thanks.
